subprojects/gdk_rust/gdk_electrum/src/lib.rs

```diff
@@ -212,6 +228,17 @@ impl ElectrumSession {
             self.wallet.as_mut().ok_or_else(|| Error::Generic("wallet not initialized".into()))?;
         Ok(wallet.write().unwrap())
     }
+
+    pub fn build_request_agent(&self) -> Result<ureq::Agent, Error> {
```

Comment:
Perhaps the agent could be reused within the session? This will reuse the same connection pool and might provide some performance gain, but I'm not sure how significant it would be.

Reply:
I checked and it seems that `ureq::get()` internally creates a new agent every time too, so I guess it probably isn't too expensive. At the very least, this doesn't degrade performance compared to before this PR.

Reply:
> this doesn't degrade performance compared to before this PR.

Yes this was the rationale for not reusing the agent.
We don't do many requests (set/get pin, asset registry, price), so it seems not worth it, at least for now.